### PR TITLE
Update branch-history script [branch-history]

### PR DESCRIPTION
**Pull request by pazner**
Add new branch-history script to try to reduce the number of false positives resulting from modification of large files (such as mesh.cpp, fespace.cpp, etc.).

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1309](https://github.com/mfem/mfem/pull/1309) | @pazner | @tzanio | @tzanio + @adrienbernede | 02/26/20 | 03/12/20 | 03/12/20 | |

### CODE REVIEW

File: tests/scripts/branch-history [hunk 1]
```diff
@@ -11,205 +11,104 @@
...
+
+# Print SHA in blue
+sub print_sha {
+  printf STDERR "\033[34m    ^---> SHA: %s\033[0m\n", $_[0];
+}
+
+# Loop over each commit in this branch, and check for large diffs
+my $total_size = 0;
+foreach my $sha (@commits) {
+  my @blobs = `git diff-tree -r -M -C --no-commit-id $sha`;
```
Comment (adrienbernede) [inline tests/scripts/branch-history]: I would suggest using long version for options:
`git diff-tree -r --find-renames --find-copies --no-commit-id $sha`

File: tests/scripts/branch-history [hunk 2]
```diff
@@ -11,205 +11,104 @@
...
-   Example usage: $0 ../..
-                  $0 ../../ -commit 8d50f48
-                  $0 ../../ -range eb2017b..5dd5511
-                  $0 ../../ -range master..d5f6f7e
-                  $0 ../../ -commit d5f6f7e -fast > /dev/null
+my $branch_max_commits = 100;
+my $commit_max_files_changed = 100;
+my $max_blob_kb = 200;
+my $max_commit_kb = 400;
+my $max_branch_kb = 1000;
```
Comment (adrienbernede) [inline tests/scripts/branch-history]: These values will have to be updated to match #1267

File: tests/scripts/branch-history (hunk 1 above)
Comment (pazner) [reply tests/scripts/branch-history]: Agreed. I think some of these flags are actually not needed, so removing for now.

File: tests/scripts/branch-history (hunk 2 above)
Comment (pazner) [reply tests/scripts/branch-history]: Good point. We should also think about good values for the blob, commit, and branch diff size limits.

**Comment by adrienbernede** [comment]
@pazner Tested on AIR_pr pathological case study. Works well.
I like the colors, although some white is missing to really mark the "made in USA" ;)

@tzanio my sentiment was that you preferred the commit ID and the message to be on the same line, don’t you? (Maybe in that case using the short SHA would help).

Now looking at @pazner comment on blob size, commit size, etc.

**Comment by tzanio** [comment]
@adrienbernede, can't access the link above

**Comment by adrienbernede** [comment]
@tzanio, should work now.

**Comment by tzanio** [comment]
@pazner, is this "ready for review"?

**Comment by pazner** [comment]
@tzanio yes, I think it's ready to be reviewed!

**Comment by adrienbernede** [comment]
@pazner, are you sure you check-marked the 55 steps of the PR checklist?
(Just kidding, currently reviewing contributing.md :P)

**Comment by tzanio** [comment]
Adding a bit more time to review...

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Comment by v-dobrev** [comment]
There is an issue with PR: now that it is merged master, the travis build of master fails: see e.g. https://travis-ci.org/github/mfem/mfem/builds/661839679. Branch builds are failing too: e.g. https://travis-ci.org/github/mfem/mfem/builds/662279594

**Comment by tzanio** [comment]
> There is an issue with PR: now that it is merged master, the travis build of master fails: see e.g. https://travis-ci.org/github/mfem/mfem/builds/661839679. Branch builds are failing too: e.g. https://travis-ci.org/github/mfem/mfem/builds/662279594

See https://github.com/mfem/mfem/pull/1286/commits/424fd7e9faf866df9956dbd769bda66ba5ad7325

**Pull request merged**